### DISCUSSION TITLE
VS compiling error: 'x_type' undeclared identifier

### DISCUSSION THREAD

**Discussion by 330305020** [SOCI/soci#1073] [Q&A] [answered]
Hello guys. 
I have VS compiling error: 'x_type' undeclared identifier. What might be the reason? I've already implement below in header file.

```cpp
struct aa
{
  std::string tt;
  int ii;
};

namespace soci
{
  // name-based conversion
  template<> struct type_conversion<aa>
  {
    typedef values base_type;

    static void from_base(values const &v, indicator /* ind */, aa &p)
    {
      // ignoring possibility that the whole object might be NULL

      p.tt = v.get<std::string>("tt");
      p.ii = v.get<int>("ii");
    }

    static void to_base(aa const & p, values & v, indicator & ind)
    {
      v.set("tt", p.tt);
      v.set("ii", p.ii);
    }
  };
}
```

**Comment by zann1x** [comment, marked as answer]
The error sounds like you're trying to use a type that has no supported conversion method. But based on the code snippet you provided, I can't see where that should be the case. You'd need to provide the context in which/how you're calling this `type_conversion` for further help on this.

**Reply by 330305020** [reply]
Thanks for your reply. I found the problem. The problem is that type of argument for use function is wrong, there's no type conversion for type char. Below is wrong:
aa A_1;
const char* name_1="James";
pSql << "SELECT * FROM test WHERE name= :id", use(name_1), into(A_1);

I should do it like below, using std::string instead of char:
aa A_1;
const std::string name_1 = "James";
pSql << "SELECT * FROM test WHERE name= :id", use(name_1), into(A_1);

Then VS build is OK.
